infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -6,29 +8,48 @@
 from datetime import datetime, timedelta
 from pathlib import Path
 from textwrap import dedent
+from typing import List, Optional, Union
 
 #####################
 # --- Constants --- #
 #####################
 
+# AWS credentials used to read/write to the database backup bucket
+AWS_ACCESS_KEY_ID = os.environ["AWS_ACCESS_KEY_ID"]
+AWS_SECRET_ACCESS_KEY = os.environ["AWS_SECRET_ACCESS_KEY"]
+# Backup specific variables
+BACKUP_BUCKET = os.environ["BACKUP_BUCKET"]
 BACKUP_DIR = Path(os.environ["BACKUP_DIR"])
 BACKUP_INTERVAL = os.environ["BACKUP_INTERVAL"]
+# Postgres connection string
 CONNECTION_STRING = "postgresql://{username}:{password}@{host}:{port}/{database}".format(
     username=os.environ["POSTGRES_USER"],
     password=os.environ["POSTGRES_PASSWORD"],
     host=os.environ["POSTGRES_HOST"],
     port=os.environ["POSTGRES_PORT"],
     database=os.environ["POSTGRES_DB"],
 )
+# Format for backup filenames
 FILENAME_DATETIME_FORMAT = "rootski-db-%m-%d-%Y_%Hh-%Mm-%Ss.sql.gz"
 
 ############################
 # --- Helper Functions --- #
 ############################
 
 
-def parse_timedelta(time_str):
-    """Parse strings of the form "1d12h" or "1h30m" or "70s" into timedelta objects."""
+class DatabaseBackupNotFoundError(Exception):
+    """Raised when when the specified backup object is not found in the S3 backups bucket."""
+
+
+def parse_timedelta(time_str: str) -> timedelta:
+    """Parse strings of the form "1d12h" or "1h30m" or "70s" into timedelta objects.
+
+    :param time_str: the string to be parsed into a :class:timedelta object, defaults to None
+    :type time_str: str
+
+    :return: returns a :class:'timedelta' object
+    :rtype: timedelta
```

Comment:
```suggestion
```

Sorry, I should write this more clearly in the style guide. Basically, we don't want to document the types in two places. We only document the types using "type annotations" like `thing: str`. This is a good option of the two ways to document because it helps with autocompletion. The parameters and return value in the docstring are for qualitative descriptions of what they do.